### PR TITLE
docs: update readme with Developing Features using Local Versions

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Enhance  docs for  Developing Features using Local Versions of palette.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,43 @@ yarn start
 
 And then either open Xcode and run, or Android Studio, or run `yarn ios` or `yarn android` from the command line, or just open up the simulator if you have done this before.
 
+## Developing Features using Local Versions of Palette
+
+When developing new components in Palette, it's often useful to test those components in consuming apps (such as Eigen). However, due to the poor support for symlinks in React Native, this can be difficult. Enter [yalc](https://github.com/wclr/yalc). Yalc is a mini package manager that one can publish to and install from, which makes it easy to test code in realtime from outside of your app.
+
+> Note: [@artsy/palette-mobile](https://github.com/artsy/palette-mobile) uses Storybooks for developing features; work there first! Then, when ready (and if necessary), test your code locally using the flow described below. You can also publish npm canary releases from the palette-mobile repo by attaching a `canary` label to your PR.
```
Comment (anandaroop) [inline README.md:61]: **thought (non-blocking):** I was actually wondering about this. Yesterday I tried to launch a Storybook server within palette-mobile for the `Image` component, but I had no idea how. I think we might need docs around that too? (Not blocking this PR of course.)

**Review by anandaroop — APPROVED** [review]
Awesome, thanks!

File: README.md (hunk 1 above)
Comment (gkartalis) [reply README.md:61]: hmm, we have this section for running palette-mobile [here](https://github.com/artsy/palette-mobile#local-development). 

When running `yarn start` in palette-mobile we start the storybook server and with `yarn android` or `yarn ios` (or even by building via xcode) you can run the storybook server.

Will merge this one but happy to hear any suggestions on making the above more clear

**Comment by artsyit** [comment]
:rocket: PR was released in [`v13.0.9`](https://github.com/artsy/palette-mobile/releases/tag/v13.0.9) :rocket:

**Comment by damassi** [comment]
<3

File: README.md (hunk 1 above)
Comment (anandaroop) [reply README.md:61]: Ah yep, those instructions do actually work once you actually read them 🤪

**Pull request merged**